Mathlib/Analysis/Normed/Module/TransferInstance.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2025 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+module
+
+public import Mathlib.Analysis.Normed.Module.Basic
+public import Mathlib.Algebra.Group.TransferInstance
+public import Mathlib.Algebra.Module.TransferInstance
+
+/-!
+# Transfer algebraic structures across `Equiv`s
+
+In this file, we transfer a (pseudo-)metric space, (semi-)normed additive commutive group
+and normed space structures across an equivalence.
+This continues the pattern set in `Mathlib/Algebra/Module/TransferInstance.lean`.
+-/
+
+@[expose] public section
+
+variable {α β : Type*}
+
+namespace Equiv
+
+variable (e : α ≃ β)
+
+/-- Transfer a `Dist` across an `Equiv` -/
+protected abbrev dist (e : α ≃ β) : ∀ [Dist β], Dist α := ⟨fun x y ↦ dist (e x) (e y)⟩
+
+/-- Transfer a `PseudoMetricSpace` across an `Equiv` -/
+protected abbrev pseudometricSpace (e : α ≃ β) : ∀ [PseudoMetricSpace β], PseudoMetricSpace α :=
+  .induced e ‹_›
+
+/-- Transfer a `MetricSpace` across an `Equiv` -/
+protected abbrev metricSpace (e : α ≃ β) : ∀ [MetricSpace β], MetricSpace α :=
+  .induced e e.injective ‹_›
```

Comment:
It feels like these belong in their own file inside `Topology/MetricSpace`, and maybe we should add the `EMetric` variants, although the latter is okay to wait for another PR when we actually need them. Maybe they don't come up often enough to matter.

Reply:
Fair point. I have moved them to a new file `Topology/MetricSpace/TransferInstance.lean`. I've chosen to not add the `EMetricSpace` versions for now: most applications will be fine with using .induced, and we can always add them later if we need them.